src/WebJobs.Extensions.DurableTask/Scale/ScaleUtils.cs

```
@@ -0,0 +1,118 @@
+            public NoOpScaleMonitor(string name, string functionId)
+            {
+#if FUNCTIONS_V3_OR_GREATER
+                this.Descriptor = new ScaleMonitorDescriptor(name, functionId);
+#else
+#pragma warning disable CS0618 // Type or member is obsolete
+                this.Descriptor = new ScaleMonitorDescriptor(name);
+#pragma warning restore CS0618 // Type or member is obsolete
+#endif
+            }
+
+            /// <summary>
+            /// A descriptive name.
+            /// </summary>
+            public ScaleMonitorDescriptor Descriptor { get; private set; }
+
+            /// <inheritdoc/>
+            Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+            {
+                throw new InvalidOperationException("The current DurableTask backend configuration does not support runtime scaling");
+            }
+
+            /// <inheritdoc/>
+            ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+            {
+                throw new InvalidOperationException("The current DurableTask backend configuration does not support runtime scaling");
+            }
+        }
+#endif
+
+#if FUNCTIONS_V3_OR_GREATER
+        internal ITargetScaler GetTargetScaler(DurabilityProvider durabilityProvider, string functionId, FunctionName functionName, string connectionName, string hubName)
+        {
+            if (durabilityProvider.TryGetTargetScaler(
+                    functionId,
+                    functionName.Name,
+                    hubName,
+                    connectionName,
+                    out ITargetScaler targetScaler))
+            {
+                return targetScaler;
+            }
+            else
+            {
+                // the durability provider does not support target-based scaling.
+                // Create an empty target scaler to avoid exceptions (unless target-based scaling is actually turned on).
+                return new NoOpTargetScaler(functionId);
+            }
```

unknown: should we "fail fast" on the scalehost if TBS is not supported?

---

Yeah I don't disagree, although I don't have a ScaleController-provided ILogger instance handy in this class. Let me try to get some clarity on this edge case from the Sc team